packages/connect-core/src/utils/descriptor/decode.ts

```diff
@@ -1,19 +1,13 @@
+import { StepDecoded } from '../../types'
 import { isCallScript, decodeCallScript } from '../callScript'
 import { isValidForwardCall, parseForwardCall } from '../forwarding'
-import { Transaction } from '../transactions'
-
-export interface TransactionWithChildren extends Transaction {
-  children?: TransactionWithChildren[]
-}
 
 /**
  * Decodes an EVM callscript and returns the transaction path it describes.
  *
  * @return An array of Ethereum transactions that describe each step in the path
  */
-export function decodeTransactionPath(
-  script: string
-): TransactionWithChildren[] {
+export function decodeForwardingPath(script: string): StepDecoded[] {
```

Comment:
Giving an evm script we convert it to a list of `StepDecoded` that we can then use to construct a `ForwardingPathDecription`.